### ISSUE TITLE
Lines.Add / Lines.Insert do not respect CRLF in a string

### ISSUE THREAD

**Issue by code-kungfu** [TurboPack/SynEdit#235]
Digging through the SynEdit code, it seems the underlying logic responsible for inserting content into the edit control, are not respecting a string with CRLF in it and shifts to a new line.

Given the follwing example:

```pascal
procedure TForm1.Button1Click(Sender: TObject);
begin
  SynEdit1.Lines.Add('This is a ' + sLineBreak + 'two line ' + sLineBreak + #32 + #32 + 'indented line' + sLineBreak);
end;
````
This will result in only the first line with *This is a * added and the rest of the content is left out.

In another practical example, if you have a string containing prettified JSON you need to display with highlighter, it will not insert this correctly at all.

I suspect it is the logic in TSynEditStringList.InsertStrings in unit SynEditTextBuffer that needs to be looked at.

A crude way I solve it for now, is with a class helper which loops through the string, splits it up every time a LineFeed control character is hit and inserts it into the SynEdit control, however this shouldn't be necessary in the first place.

**Comment by chapmanworld** [comment]
Another work around...

 SynEdit1.Lines.Text := SynEdit1.Lines.Text + 'This is a ' + sLineBreak + 'two line ' + sLineBreak + #32 + #32 + 'indented line' + sLineBreak;

**Comment by pyscripter** [comment]
SynEdit.Lines inherits from TStrings and behaves the same.

Try this 
```pascal
  var SL := TStringList.Create;
  SL.Add('This is a ' + sLineBreak + 'two line ' + sLineBreak + #32 + #32 + 'indented line' + sLineBreak);
  ShowMessage(SL.Text);
  ShowMessage(SL.Count.ToString);
  SL.Free;
```
You get **one line** with the text which includes line breaks.   

Now try with SynEdit assuming you have SynEdit1:

```pascal
  SynEdit1.Lines.Clear;
  SynEdit1.Lines.Add('This is a ' + sLineBreak + 'two line ' + sLineBreak + #32 + #32 + 'indented line' + sLineBreak);
  ShowMessage(SynEdit1.Text);
  ShowMessage(SynEdit1.Lines.Count.ToString);
```

You get exactly the same.   Your full  line is there (!) but only the first part is displayed.   This is because each visually displayed line corresponds to line of the underlying TStrings descendent and cannot take more vertical space.

The way to do it is:

  var S: string := 'This is a ' + sLineBreak + 'two line ' + sLineBreak + #32 + #32 + 'indented line' + sLineBreak;
  SynEdit1.Lines.AddStrings(S.Split([sLineBreak]));

**Comment by code-kungfu** [comment]
With all due respect, but it is a design defect and it isn't working as designed when you look at every other edit control's behavior which are using the same Lines derivative approach.

I'm happy with either a) fix the behavior of Lines.Add() or b) add a method to the SynEdit base class which is more intuitive to use, like SynEdit1.AddLines() which takes care of the housekeeping or something along those lines.

**Comment by pyscripter** [comment]
> b) add a method to the SynEdit base class which is more intuitive to use, like SynEdit1.AddLines()

SynEdit.Lines.InsertText?

**Comment by code-kungfu** [comment]
> The SynEdit text buffer is manipulated through its Lines property which is a TStrings descendent.
Yes, but the behavior is not correct compared to any other editor control available, and that is a defect. 
Lines.Add() has always handled a string with multiple CRLF occurrences correctly.



> SynEdit.Lines.InsertText?
It is not intuitive.

**Comment by pyscripter** [comment]
TSynEditStringList provides a public InsertText method exactly for this reason.  It splits the line and inserts the result into the string list.  However to use it you need to typecast.

`TSynEditStringList(SynEdit1.Lines).InsertText(Index, NewText);`
